core/src/main/scala/com/microsoft/azure/synapse/ml/core/utils/MapUtils.scala

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.core.utils
+
+object MapUtils {
+  def newMapArray(args: Seq[Seq[Any]]): Array[Map[Any, Any]] = {
+    val result = args.map(a => newMap(a))
+    println(result)
+    result.toArray
+  }
+
+  def newMap(args: Seq[Any]): Map[Any, Any] = {
+    if (args.length % 2 != 0) {
+      throw new IllegalArgumentException("Argument list must have an even number of elements.")
+    }
+
+    val map = scala.collection.mutable.Map[Any, Any]()
+    addArgs(map, args)
+  }
+
+  def addArgs(map: scala.collection.mutable.Map[Any, Any], args: Seq[Any]): Map[Any, Any] = {
+    if (args != null && args.length > 1) {
+      map.put(args(0), args(1))
+      addArgs(map, args.tail.tail)
+    }
+
+    map.toMap
+  }
+}
```

Comment:
no need to makle a mutable map here, just map the Seq to a Seq or ordered pairs then call .toMap. Could probabbly do this in a single func too

Reply:
This turned out to be a dead end too. But for grins I tried your suggestion and came up with
  def newMapArray(keys: Seq[Seq[Any]], values: Seq[Seq[Any]]): Array[Map[Any, Any]] = {
    (keys zip values).map(t => (t._1 zip t._2).toMap).toArray
  }
Scala can be too concise for easy reading!